internal/service/docdb/cluster_instance.go

```diff
@@ -188,12 +200,14 @@ func resourceClusterInstanceCreate(d *schema.ResourceData, meta interface{}) err
 	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
 
 	createOpts := &docdb.CreateDBInstanceInput{
-		DBInstanceClass:         aws.String(d.Get("instance_class").(string)),
-		DBClusterIdentifier:     aws.String(d.Get("cluster_identifier").(string)),
-		Engine:                  aws.String(d.Get("engine").(string)),
-		PromotionTier:           aws.Int64(int64(d.Get("promotion_tier").(int))),
-		AutoMinorVersionUpgrade: aws.Bool(d.Get("auto_minor_version_upgrade").(bool)),
-		Tags:                    Tags(tags.IgnoreAWS()),
+		DBInstanceClass:             aws.String(d.Get("instance_class").(string)),
+		DBClusterIdentifier:         aws.String(d.Get("cluster_identifier").(string)),
+		Engine:                      aws.String(d.Get("engine").(string)),
+		PromotionTier:               aws.Int64(int64(d.Get("promotion_tier").(int))),
+		AutoMinorVersionUpgrade:     aws.Bool(d.Get("auto_minor_version_upgrade").(bool)),
+		Tags:                        Tags(tags.IgnoreAWS()),
+		EnablePerformanceInsights:   aws.Bool(d.Get("enable_performance_insights").(bool)),
+		PerformanceInsightsKMSKeyId: aws.String(d.Get("performance_insights_kms_key_id").(string)),
```

Comment:
setting `PerformanceInsightsKMSKeyId` inside the `Input` object means the key is always passed. we only want to pass this key if its not null. instead this should be conditionally added to `createOpts`